File: LeetCode/1475/lc1475.go
package lc1475

func finalPrices(prices []int) []int {
	ans := prices[:]
	for i, price := range prices {
		for _, p := range prices[i+1:] {
			if p <= price {
				ans[i] -= p
				break // inner for loop
			}
		}
	}
	return ans
}

File: LeetCode/1475/python/lc1475.py
from typing import List

# Runtime complexity: O(n^2)
# Auxiliary space complexity: O(n) (for storing the answer)
# Subjective level: easy
# Solved on: 2026-02-09
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        ans = prices[:]
        for idx in range(len(prices)):
            for idxFollowing in range(idx+1, len(prices)):
                if prices[idxFollowing] <= prices[idx]:
                    ans[idx] -= prices[idxFollowing]
                    break # the inner for idxFollowing... loop
        return ans
